Update WorkoutScheduler.jsx

Ruslan1908 opened the pull request:
1. Вынесена инициализация useState в отдельную функцию: Добавлена функция getInitialWorkoutPlan, которая:
   Загружает план из localStorage, если он есть. Генерирует новый план с помощью generateWorkoutPlan,  если данных нет.
2. Добавлена функция-обертка для обновления состояния: Создана функция setWorkoutPlan, которая одновременно  обновляет состояние и сохраняет его в localStorage.

3. generateWorkoutPlan: Объект, добавляемый в массив workoutPlan, разбит на  отдельные строки. Добавлен комментарий, поясняющий использование  оператора % для циклического выбора мышечных групп.

4. Удален useEffect: Сохранение изменений в localStorage теперь выполняется  непосредственно в setWorkoutPlan, поэтому использование  useEffect стало не нужнл.

5. изменен export

Ruslan1908 committed:
Update WorkoutScheduler.jsx

1. Вынесена инициализация useState в отдельную функцию:
Добавлена функция getInitialWorkoutPlan, которая:
   Загружает план из localStorage, если он есть.
   Генерирует новый план с помощью generateWorkoutPlan, 
   если данных нет.
2. Добавлена функция-обертка для обновления состояния:
    Создана функция setWorkoutPlan, которая одновременно 
    обновляет состояние и сохраняет его в localStorage.

3. generateWorkoutPlan:
    Объект, добавляемый в массив workoutPlan, разбит на 
    отдельные строки.
    Добавлен комментарий, поясняющий использование 
    оператора % для циклического выбора мышечных групп.

4. Удален useEffect:
   Сохранение изменений в localStorage теперь выполняется 
   непосредственно в setWorkoutPlan, поэтому использование 
   useEffect стало не нужнл.

5. изменен export

AntonZlobichLambda commented on src/WorkoutScheduler.jsx:
```diff
@@ -10,23 +9,31 @@ const generateWorkoutPlan = (sessionsPerWeek) => {
-    workoutPlan.push({ day: `День ${i + 1}`, muscles: muscleGroups[i % muscleGroups.length], completed: false });
+    workoutPlan.push({
+      day: `День ${i + 1}`, // Название дня тренировки
+      muscles: muscleGroups[i % muscleGroups.length], // Определяем мышечные группы с помощью остатка от деления
```
По поводу комментария к этому фрагменту возникло недопонимание)
Поля понятно за что отвечают (day, muscles, completed).
Здесь больше вопросов вызывает именно остаток от деления.
Мне бы наверное хотелось тут видеть как ревьюверу или как разработчику, который должен потом этот функционал как-то расширять, зачем именно остаток от деления. И твоё устное объяснение про то, что "это связано с тем, что дней может быть больше чем доступных групп мышц и тогда, чтобы продолжить цикл прохождения по всем группам мышц сначала, мы используем остаток от деления" - это как раз то, что нужно знать. Ну только мб чуть более ёмко)

AntonZlobichLambda commented on src/WorkoutScheduler.jsx:
```diff
@@ -10,23 +9,31 @@ const generateWorkoutPlan = (sessionsPerWeek) => {
+};
+
+export const WorkoutScheduler = ({ sessionsPerWeek }) => {
+  const [workoutPlan, setWorkoutPlanState] = useState(() => getInitialWorkoutPlan(sessionsPerWeek));
```
Название сеттера должно быть точно таким же как и переменной, только с set вначале. Как у тебя было `[workoutPlan, setWorkoutPlan]`. `[value, setValue]`.

Я понимаю, почему ты так сделал, но лучше вернуть `setWorkoutPlan` и переназвать ту функцию, которая ниже.

AntonZlobichLambda commented on src/WorkoutScheduler.jsx:
```diff
@@ -10,23 +9,31 @@ const generateWorkoutPlan = (sessionsPerWeek) => {
+export const WorkoutScheduler = ({ sessionsPerWeek }) => {
+  const [workoutPlan, setWorkoutPlanState] = useState(() => getInitialWorkoutPlan(sessionsPerWeek));
+
+  const setWorkoutPlan = (newPlan) => {
```
Я бы так же сделал (объединил бы под одной функцией setState и update local storag'а), молодец 👍

Ruslan1908 committed:
Update WorkoutScheduler.jsx

1. Переименовал вспомогательную функцию setWorkoutPlan в updateWorkoutPlan.

2. Вернул setWorkoutPlan для сеттера.

3. Поправил комментарий на счет процента
